client/src/main/java/com/alibaba/nacos/client/naming/core/HostReactor.java

```diff
@@ -108,6 +111,10 @@ public Thread newThread(Runnable r) {
         this.updatingMap = new ConcurrentHashMap<String, Object>();
         this.failoverReactor = new FailoverReactor(this, cacheDir);
         this.pushReceiver = new PushReceiver(this);
+        
+        NotifyCenter.registerToPublisher(InstancesChangeEvent.class, 16384);
+        notifier = new InstancesChangeNotifier();
```

Comment:
notifier = new InstancesChangeNotifier(); => this.notifier = new InstancesChangeNotifier();

Reply:
really careful